pkg/runtime/bulk_subscriber.go

```diff
@@ -429,21 +458,19 @@ func (a *DaprRuntime) publishBulkMessageHTTP(ctx context.Context, msg *pubsubBul
 		// These are errors that are not retriable, for now it is just 404 but more status codes can be added.
 		// When adding/removing an error here, check if that is also applicable to GRPC since there is a mapping between HTTP and GRPC errors:
 		// https://cloud.google.com/apis/design/errors#handling_errors
-		log.Errorf("non-retriable error returned from app while processing bulk pub/sub event. status code returned: %v", statusCode)
-		bulkSubDiag.statusWiseDiag[string(pubsub.Drop)] += int64(len(msg.entries))
-		bulkSubDiag.elapsed = elapsed
-		populateBulkSubscribeResponsesWithError(msg.entries, bulkResponses, &entryIdIndexMap, nil)
+		log.Errorf("Non-retriable error returned from app while processing bulk pub/sub event. status code returned: %v", statusCode)
+		bscData.bulkSubDiag.statusWiseDiag[string(pubsub.Drop)] += int64(len(rawMsgEntries))
+		bscData.bulkSubDiag.elapsed = elapsed
+		populateBulkSubscribeResponsesWithError(psm, bscData.bulkResponses, nil)
 		return nil
 	}
 
 	// Every error from now on is a retriable error.
-	log.Warnf("retriable error returned from app while processing bulk pub/sub event, topic: %v. status code returned: %v", msg.topic, statusCode)
-	bulkSubDiag.statusWiseDiag[string(pubsub.Retry)] += int64(len(msg.entries))
-	bulkSubDiag.elapsed = elapsed
-	populateBulkSubscribeResponsesWithError(
-		msg.entries, bulkResponses, &entryIdIndexMap,
-		fmt.Errorf("retriable error returned from app while processing bulk pub/sub event, topic: %v. status code returned: %v", msg.topic, statusCode))
-	return fmt.Errorf("retriable error returned from app while processing bulk pub/sub event, topic: %v. status code returned: %v", msg.topic, statusCode)
+	log.Warnf("Retriable error returned from app while processing bulk pub/sub event, topic: %v. status code returned: %v", psm.topic, statusCode)
+	bscData.bulkSubDiag.statusWiseDiag[string(pubsub.Retry)] += int64(len(rawMsgEntries))
+	bscData.bulkSubDiag.elapsed = elapsed
+	populateBulkSubscribeResponsesWithError(psm, bscData.bulkResponses, fmt.Errorf("retriable error returned from app while processing bulk pub/sub event, topic: %v. status code returned: %v", psm.topic, statusCode))
```

Comment:
extract as error var and reuse in line below.

Reply:
Done.